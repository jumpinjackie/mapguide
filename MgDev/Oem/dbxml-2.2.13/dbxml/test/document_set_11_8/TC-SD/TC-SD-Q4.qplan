<XQuery>
  <FLWOR>
    <LetBinding name="ent">
      <Navigation>
        <QueryPlanFunction result="collection" container="11.8.4.dbxml">
          <OQPlan>V(node-element-equality-string,hw,=,'you')</OQPlan>
        </QueryPlanFunction>
        <Step axis="child" name="dictionary" nodeType="element"/>
        <Step axis="child" name="e" nodeType="element"/>
        <DbXmlFilter>
          <Navigation>
            <Step axis="child" name="hwg" nodeType="element"/>
            <Step axis="child" name="hw" nodeType="element"/>
            <DbXmlCompare name="equal">
              <Sequence>
                <AnyAtomicTypeConstructor value="you" typeuri="http://www.w3.org/2001/XMLSchema" typename="string"/>
              </Sequence>
            </DbXmlCompare>
          </Navigation>
        </DbXmlFilter>
      </Navigation>
    </LetBinding>
    <ForBinding name="prevEnt">
      <Navigation>
        <QueryPlanFunction result="collection" container="11.8.4.dbxml">
          <OQPlan>U</OQPlan>
        </QueryPlanFunction>
        <Step axis="child" name="dictionary" nodeType="element"/>
        <Step axis="child" name="e" nodeType="element">
          <Predicates>
            <SomeFLWOR>
              <ForBinding name="a">
                <Navigation>
                  <Step axis="child" name="hwg" nodeType="element"/>
                  <Step axis="child" name="hw" nodeType="element">
                    <OQPlan>P(node-element-equality-string,prefix,hw)</OQPlan>
                  </Step>
                </Navigation>
              </ForBinding>
              <Operator name="node-order">
                <Variable name="a"/>
                <Variable name="ent"/>
              </Operator>
            </SomeFLWOR>
            <Operator name="equal">
              <Function name="{http://www.w3.org/2005/04/xpath-functions}:position"/>
              <Function name="{http://www.w3.org/2005/04/xpath-functions}:last"/>
            </Operator>
          </Predicates>
        </Step>
      </Navigation>
    </ForBinding>
    <DOMConstructor type="element">
      <Name>
        <Sequence>
          <AnyAtomicTypeConstructor value="Output" typeuri="http://www.w3.org/2001/XMLSchema" typename="string"/>
        </Sequence>
      </Name>
      <Children>
        <DOMConstructor type="element">
          <Name>
            <Sequence>
              <AnyAtomicTypeConstructor value="CurrentEntry" typeuri="http://www.w3.org/2001/XMLSchema" typename="string"/>
            </Sequence>
          </Name>
          <Children>
            <Navigation>
              <Variable name="ent"/>
              <Step axis="child" name="hwg" nodeType="element"/>
              <Step axis="child" name="hw" nodeType="element"/>
              <Step axis="child">
                <ItemType testType="text"/>
              </Step>
            </Navigation>
          </Children>
        </DOMConstructor>
        <DOMConstructor type="element">
          <Name>
            <Sequence>
              <AnyAtomicTypeConstructor value="PreviousEntry" typeuri="http://www.w3.org/2001/XMLSchema" typename="string"/>
            </Sequence>
          </Name>
          <Children>
            <Navigation>
              <Variable name="prevEnt"/>
              <Step axis="child" name="hwg" nodeType="element"/>
              <Step axis="child" name="hw" nodeType="element">
                <OQPlan>P(node-element-equality-string,prefix,hw)</OQPlan>
              </Step>
              <Step axis="child">
                <ItemType testType="text"/>
              </Step>
            </Navigation>
          </Children>
        </DOMConstructor>
      </Children>
    </DOMConstructor>
  </FLWOR>
</XQuery>
