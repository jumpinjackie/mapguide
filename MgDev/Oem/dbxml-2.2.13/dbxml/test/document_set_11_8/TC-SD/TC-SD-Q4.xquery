let $ent := collection("11.8.4.dbxml")/dictionary/e[hwg/hw="you"]
for $prevEnt in collection("11.8.4.dbxml")/dictionary/e[some $a in hwg/hw satisfies $a << $ent]
    [position() = last()]
return
    <Output>
        <CurrentEntry>{$ent/hwg/hw/text()}</CurrentEntry>
        <PreviousEntry>{$prevEnt/hwg/hw/text()}</PreviousEntry>
    </Output>
