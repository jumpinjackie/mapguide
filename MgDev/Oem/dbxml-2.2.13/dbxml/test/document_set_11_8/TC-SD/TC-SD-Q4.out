<Output><CurrentEntry>untuneyou</CurrentEntry><PreviousEntry>chinks</PreviousEntry></Output>